### PR TITLE
Add fun.kikoplay.KikoPlay

### PR DESCRIPTION
**Pull request by neovali**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:https://github.com/KikoPlayProject/KikoPlay/pull/124**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge). Author:@Protostars
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

File: fun.kikoplay.KikoPlay.yml [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+command: KikoPlay
+finish-args:
+  - --device=dri
+  - --share=ipc
+  - --share=network
+  # aria2c download
+  - --filesystem=xdg-videos
+  - --filesystem=xdg-download
+  - --socket=wayland
+  - --socket=fallback-x11
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:15]: the patch to the .desktop file indicate this doesn't support Wayland.

File: fun.kikoplay.KikoPlay.yml [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  - --socket=pulseaudio
+  # for hardware acceleration
+  - --device=dri
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:18]:
```suggestion
```

it's already on line 8

File: fun.kikoplay.KikoPlay.yml [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+          stable-only: true
+          url-template: https://github.com/libass/libass/releases/download/$version/libass-$version.tar.gz
+
+  - name: libv4l2
+    cleanup:
+      - /sbin
+      - /bin
+      - /include
+      - /lib/*.la
+      - /lib*/*/*/*.la
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:57]:
```suggestion
      - "*.la"
```

you also probably need "*.a"

File: fun.kikoplay.KikoPlay.yml [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+cleanup-commands:
+  - mkdir -p ${FLATPAK_DEST}/lib/ffmpeg
+
+modules:
+  - shared-modules/luajit/luajit.json
+
+  - name: libass
+    cleanup:
+      - /include
+      - /lib/*.la
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:37]:
```suggestion
      - "*.la"
```

you also probably need "*.a"

File: fun.kikoplay.KikoPlay.yml [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  
+  - name: kikoplay-app
+    buildsystem: simple
+    build-options:
+      no-debuginfo: true
+    build-commands: 
+      - cp -a app $FLATPAK_DEST/bin/extension/
+    sources: 
+      - type: git
+        url: https://github.com/KikoPlayProject/KikoPlayApp.git
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:230]: a commit or tag is required.

File: fun.kikoplay.KikoPlay.yml [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+    build-commands: 
+      - mkdir -p $FLATPAK_DEST/bin/extension/script
+      # script
+      - cp -a bgm_calendar $FLATPAK_DEST/bin/extension/script
+      - cp -a danmu $FLATPAK_DEST/bin/extension/script
+      - cp -a library $FLATPAK_DEST/bin/extension/script
+      - cp -a resource $FLATPAK_DEST/bin/extension/script
+    sources:
+      - type: git
+        url: https://github.com/KikoPlayProject/KikoPlayScript.git
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:220]: a commit or tag is required.

File: fun.kikoplay.KikoPlay.yml [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+      - MYCFLAGS=-fPIC
+      - linux
+    make-install-args:
+      - INSTALL_TOP=/app
+    sources:
+      - type: archive
+        url: https://www.lua.org/ftp/lua-5.3.6.tar.gz
+        sha256: fc5fd69bb8736323f026672b1b7235da613d7177e72558893a0bdcd320466d60
+
+  - name: resources
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:194]: this should be part of the main module `KikoPlay`

File: patch/0001-prebuild.patch [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+ rename Extension/Lua/{lua.pro => Lua.pro} (100%)
+
+diff --git a/Extension/Lua/lua.pro b/Extension/Lua/Lua.pro
+similarity index 100%
+rename from Extension/Lua/lua.pro
+rename to Extension/Lua/Lua.pro
+diff --git a/lib/x64/linux/liblua53.a b/lib/x64/linux/liblua53.a
+index d5dc920..bf39146 120000
+--- a/lib/x64/linux/liblua53.a
++++ b/lib/x64/linux/liblua53.a
```
Comment (hfiguiere) [inline patch/0001-prebuild.patch:19]: does this work on aarch64?

Comment (neovali) [reply patch/0001-prebuild.patch:19]: no

File: fun.kikoplay.KikoPlay.yml [hunk 9]
```diff
@@ -53,8 +50,7 @@ modules:
       - /sbin
       - /bin
       - /include
-      - /lib/*.la
-      - /lib*/*/*/*.la
+      - "*.a"
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:53]: what about the .la?

File: fun.kikoplay.KikoPlay.yml [hunk 10]
```diff
@@ -241,3 +226,16 @@ modules:
...
+        build-options:
+          no-debuginfo: true
+        build-commands:
+          - install -Dm644 fun.kikoplay.KikoPlay.metainfo.xml -t $FLATPAK_DEST/share/metainfo
+          - install -Dm644 fun.kikoplay.KikoPlay.png -t $FLATPAK_DEST/share/icons/hicolor/128x128/apps
+        sources:
+          - type: file
+            path: fun.kikoplay.KikoPlay.metainfo.xml
+          - type: file
+            path: fun.kikoplay.KikoPlay.png
```
Comment (hfiguiere) [inline fun.kikoplay.KikoPlay.yml:241]:
```suggestion
          - type: file
            path: fun.kikoplay.KikoPlay.metainfo.xml
          - type: file
            path: fun.kikoplay.KikoPlay.png
        post-install:
          - install -Dm644 fun.kikoplay.KikoPlay.metainfo.xml -t $FLATPAK_DEST/share/metainfo
          - install -Dm644 fun.kikoplay.KikoPlay.png -t $FLATPAK_DEST/share/icons/hicolor/128x128/apps
```


this is what I meant

File: patch/0001-prebuild.patch (hunk 8 above)
Comment (hfiguiere) [reply patch/0001-prebuild.patch:19]: can't this be just built?

File: fun.kikoplay.KikoPlay.yml (hunk 10 above)
Comment (neovali) [reply fun.kikoplay.KikoPlay.yml]: sorry.

File: patch/0001-prebuild.patch (hunk 8 above)
Comment (neovali) [reply patch/0001-prebuild.patch:19]: it seems that [upstream](https://github.com/KikoPlayProject/KikoPlay/tree/master/lib/x64) only provides x64 libraries

Comment (neovali) [reply patch/0001-prebuild.patch:19]: fine, the author said "I have never tried it before." 
maybe i works

**Comment by neovali** [comment]
bot, build

**Comment by neovali** [comment]
i will try rename to io.github.kikoplay.Kikoplay

**Pull request closed without merging**